tool/tsh/tsh.go

```diff
@@ -2846,6 +2858,15 @@ func printProfiles(debug bool, profile *client.ProfileStatus, profiles []*client
 	for _, p := range profiles {
 		printStatus(debug, p, false)
 	}
+
+	// Print relevant active env vars, if they are set.
+	env := getStatusEnv()
+	if len(env) > 0 {
```

Comment:
@GavinFrazar @ravicious Couple of notes here:

- Is this going to be printed by default? I'm not sure we'd want that, this should probably be hidden behind a `-v` flag (we should have verbose flag already).
- I'm not sure why we don't want to just show the cluster name from TELEPORT_CLUSTER variable in the tsh status by default? It's supposed to show the currently active cluster so if you have the env var set, then it makes sense to show it. Showing both might only be more confusing for users IMO.

I think that displaying set env. vars with -v flag is a good idea, but I would go back to the original suggestion and also showed the actual cluster name in tsh status.

Reply:
I did that originally, but thought that it might be cleaner to just show the profile and environment separately. Should we add an indicator that the cluster is being selected by the env var, like "(from TELEPORT_CLUSTER)" next to the name?

Reply:
Sure, that sounds good. You can also "pin" a kube cluster same way using `TELEPORT_KUBE_CLUSTER` variable, let's also do the same for it while we're here.

Reply:
For `$ tsh status` it's fine to display `(TELEPORT_CLUSTER is set)` next to the active profile cluster, but what if they specify `$ tsh status --format=json` or use `--format=yaml`? I don't know if we want to put that info in there still, since it may be consumed by some other program by a user (I have no idea whether anyone actually does this).

This makes me want to lean towards just displaying the info separately beneath the profile(s), or in yaml/json, adding an "environment:" entry. We can still gate this new behavior behind a "-v --verbose" flag, which tsh status/login does not have currently.

Examples:
```
> tsh status -v                                                                                     (base) 
> Profile URL:        https://proxy1.local.gd:3080
  Logged in as:       bob
  Cluster:            cluster2
  Roles:              access, auditor, editor
  Logins:             gavin, -teleport-internal-join
  Kubernetes:         enabled
  Valid until:        2022-06-02 00:43:03 -0700 PDT [valid for 11h20m0s]
  Extensions:         permit-agent-forwarding, permit-port-forwarding, permit-pty

Active Environment
TELEPORT_SITE=cluster2
```

```
> tsh status -v --format=yaml                                                                       (base) 
active:
  cluster: cluster1
  extensions:
  - permit-agent-forwarding
  - permit-port-forwarding
  - permit-pty
  kubernetes_enabled: true
  logins:
  - gavin
  - -teleport-internal-join
  profile_url: https://proxy1.local.gd:3080
  roles:
  - access
  - auditor
  - editor
  traits:
    logins:
    - gavin
  username: bob
  valid_until: "2022-06-02T00:43:03-07:00"
environment:
  TELEPORT_CLUSTER: cluster1
  TELEPORT_PROXY: proxy1.local.gd:3080
profiles: []
```

Reply:
I've made the behavior this:

`$ tsh status` displays the active profile with proxy, cluster, and kube cluster settings from env variables, if they are set. It will display the env var shadowing the active profile setting next to the output in parentheses.

A new verbose flag (-v, --verbose) can be specified to display the active environment below the profiles.

Example output:
```fish
> tsh status -v                                                                                     (base) 
> Profile URL:        https://proxy1.local.gd:3080 (TELEPORT_PROXY)
  Logged in as:       alice
  Cluster:            cluster2 (TELEPORT_CLUSTER)
  Roles:              access, auditor, editor
  Logins:             gavin, -teleport-internal-join
  Kubernetes:         enabled
  Kubernetes cluster: "kubeSTUB (TELEPORT_KUBE_CLUSTER)"
  Valid until:        2022-06-07 02:44:33 -0700 PDT [valid for 11h55m0s]
  Extensions:         permit-agent-forwarding, permit-port-forwarding, permit-pty

Active Environment
TELEPORT_CLUSTER=cluster2
TELEPORT_KUBE_CLUSTER=kubeSTUB
TELEPORT_PROXY=proxy1.local.gd:3080
```

And json output example:
```fish
{
  "active": {
    "profile_url": "https://proxy1.local.gd:3080",
    "username": "alice",
    "cluster": "cluster2",
    "roles": [
      "access",
      "auditor",
      "editor"
    ],
    "traits": {
      "aws_role_arns": null,
      "db_names": null,
      "db_users": null,
      "kubernetes_groups": null,
      "kubernetes_users": null,
      "logins": [
  "gavin"
],
      "windows_logins": null
    },
    "logins": [
      "gavin",
      "-teleport-internal-join"
    ],
    "kubernetes_enabled": true,
    "kubernetes_cluster": "kubeSTUB",
    "valid_until": "2022-06-07T02:44:33-07:00",
    "extensions": [
      "permit-agent-forwarding",
      "permit-port-forwarding",
      "permit-pty"
    ]
  },
  "profiles": [],
  "environment": {
    "TELEPORT_CLUSTER": "cluster2",
    "TELEPORT_KUBE_CLUSTER": "kubeSTUB",
    "TELEPORT_PROXY": "proxy1.local.gd:3080"
  }
}
```

Reply:
@r0mant can we mark this resolved?